intezer_sdk/api.py

```diff
@@ -63,20 +67,40 @@ def analyze_by_hash(self,
         return self._get_analysis_id_from_response(response)
 
     def analyze_by_file(self,
-                        file_path,
-                        dynamic_unpacking=None,
-                        static_unpacking=None):  # type: (str,bool,bool) -> str
+                        file_path: str = None,
+                        file_stream: typing.BinaryIO = None,
+                        dynamic_unpacking: bool = None,
+                        static_unpacking: bool = None) -> str:
+        if file_path:
+            file_context_manager = open(file_path, 'rb')
+            file_name = os.path.basename(file_path)
+        elif file_stream:
+            file_context_manager = dummy_context_manager(file_stream)
```

Comment:
i don't like it... have a function that performs an analysis on a stream and reuse it when file path is received

Reply:
Done